qiskit/ignis/verification/topological_codes/fitters.py

```diff
@@ -159,46 +159,91 @@ def _make_syndrome_graph(self):
 
         return S
 
-    def weight_syndrome_graph(self, results):
-        """Generate weighted syndrome graph from result counts.
+    def get_error_probs(self, results):
+        """
+        Generate probabilities of single error events from result counts.
 
         Args:
             results (dict): A results dictionary, as produced by the
             `process_results` method of the code.
 
+        Returns:
+            dict: Keys are the edges for specific error
+            events, and values are the calculated probabilities
+
         Additional information:
             Uses `results` to estimate the probability of the errors that
-            create the pairs of nodes in S. The edge weights are then
-            replaced with the corresponding -log(p/(1-p).
+            create the pairs of nodes specified by the edge.
+            Calculation done using the method of Spitz, et al.
+            https://doi.org/10.1002/qute.201800012
         """
 
         results = results['0']
+        shots = sum(results.values())
+
+        nodes = self.S.nodes()
 
-        count = {element: {edge: 0 for edge in self.S.edges}
-                 for element in ['00', '01', '10', '11']}
+        error_probs = {}
+        for edge in self.S.edge_list():
 
-        for string in results:
+            # initialize averages
+            av_vv = 0  # v_ij
+            av_v = [0, 0]  # [v_,v_j]
+            av_xor = 0  # v_{i xor j}
 
-            nodes = self._string2nodes(string)
+            for string in results:
 
-            for edge in self.S.edge_list():
-                element = ''
-                for j in range(2):
-                    if edge[j] in nodes:
-                        element += '1'
-                    else:
-                        element += '0'
-                count[element][edge] += results[string]
+                # list of i for which v_i=1
+                error_nodes = self._string2nodes(string)
+
+                # get [v_i,v_j] for edge (i,j)
+                v = [int(nodes[edge[k]] in error_nodes) for k in range(2)]
+
+                # update averages
+                av_vv += v[0]*v[1]*results[string]
+                for k in range(2):
+                    av_v[k] += v[k]*results[string]
+                av_xor += (v[0] != v[1])*results[string]
+
+            # normalize
+            av_vv /= shots
+            av_v[0] /= shots
+            av_v[1] /= shots
+            av_xor /= shots
+
+            if (1 - 2*av_xor) != 0:
+                x = (av_vv - av_v[0]*av_v[1])/(1 - 2*av_xor)
+            else:
+                x = np.nan
+            error_probs[nodes[edge[0]], nodes[edge[1]]] = max(0, 0.5 - np.sqrt(0.25-x))
```

Comment:
Same as above to get the node's weight/payload I would just get it from the syndrome graph directly:

```suggestion
            error_probs[self.S[edge[0]], self.S[edge[1]]] = max(0, 0.5 - np.sqrt(0.25-x))
```